### ISSUE TITLE
Fix Musl compatibilty (malloc_trim)

### ISSUE THREAD

**Issue by bivsk** [ilya-zlobintsev/LACT#352]
### Checklist

- [X] The problem is not listed in the [hardware support matrix](https://github.com/ilya-zlobintsev/LACT?tab=readme-ov-file#hardware-support) as a known limitation
- [X] I've included a [debug snapshot](https://github.com/ilya-zlobintsev/LACT?tab=readme-ov-file#reporting-issues) if the issue is related to hardware configuration

### Bug description

the malloc_trim glibc extension is not available on musl, causing builds to fail.  It would be nice to add support for musl systems.

```
error[E0425]: cannot find function `malloc_trim` in crate `libc`
   --> lact-daemon/src/server/handler.rs:144:19
    |
144 |             libc::malloc_trim(0);
    |                   ^^^^^^^^^^^ not found in `libc`
```

### System info

```markdown
- Distribution: Gentoo (musl)
```

**Comment by ilya-zlobintsev** [comment]
Please try the `fix/musl` branch, https://github.com/ilya-zlobintsev/LACT/commit/d59b04c3ccb09e8bd19782f7a7cf40d145a69f41 should fix it.

**Comment by bivsk** [comment]
> Please try the `fix/musl` branch, [d59b04c](https://github.com/ilya-zlobintsev/LACT/commit/d59b04c3ccb09e8bd19782f7a7cf40d145a69f41) should fix it.

Looks all fixed on my end! thanks!

**Comment by ilya-zlobintsev** [comment]
Fixed in https://github.com/ilya-zlobintsev/LACT/pull/353
